app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,85 @@
+class RecipesController < ApplicationController
+  # before_action :authenticate_user!, except: [:show]
+  include RecipesHelper
+
+  def index
+    @recipes = Recipe.includes(:recipe_foods).where(user_id: current_user.id)
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def toggle_public
+    @recipe = Recipe.find(params[:id])
+    @recipe.toggle!(:public)
+    respond_to do |format|
+      if @recipe.save
+        format.html { redirect_to recipes_path, notice: 'Recipe was successfully updated.' }
+        format.json { render :show, status: :created, location: @recipe }
+      else
+        format.html { redirect_to recipes_path, alert: 'Recipe was not updated.' }
+        format.json { render json: @recipe.errors, status: unprocessable_entity }
+      end
+    end
+  end
+
+  def remove_food_from_recipe
+    @recipe_food = RecipeFood.includes(:food).find_by(recipe_id: params[:id], food_id: params[:food_id])
+    @recipe_food.destroy
+    redirect_to recipe_path(params[:id])
+  end
+
+  def public_recipes
+    @users = User.includes(:recipes).all
+    @foods = Food.includes(:recipe_foods).all
+    @public_recipes = Recipe.includes(:user, :recipe_foods).where(public: true).order(created_at: :desc)
+    @current_user = current_user
+  end
+
+  def create
+    @recipe = current_user.recipes.build(recipe_params)
```

Comment:
- [ ] Nothing happens when details are entered for a recipe. Like indicated in the previous review, let's ensure to improve upon this aspect.